File: app/src/main/java/com/printf/kidsteacher/fragment/BaseFragment.java
package com.printf.kidsteacher.fragment;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.gson.Gson;

public class BaseFragment extends Fragment
{
    public Gson gson;
    protected Context activity;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        gson = new Gson();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        activity = context;
    }
}

File: app/src/main/java/com/printf/kidsteacher/fragment/BaseFragment.kt
package com.printf.kidsteacher.fragment

import android.content.Context
import com.google.gson.Gson
import android.os.Bundle
import androidx.fragment.app.Fragment

open class BaseFragment : Fragment() {
    var gson: Gson? = null

    @JvmField
    protected var activity: Context? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        gson = Gson()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        activity = context
    }
}